src/IssueViz.Security/Taint/TaintSyncPackage.cs

```diff
@@ -55,6 +55,8 @@ public sealed class TaintSyncPackage : AsyncPackage
 
         protected override async Task InitializeAsync(CancellationToken cancellationToken, IProgress<ServiceProgressData> progress)
         {
+            await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
```

Comment:
This is good, because we need to make the `GetService` calls on the UI thread.

However, I think we can push some of the later work to the background thread e.g. 

```
// Switch to UI thread
// Fetch the componentModel and logger services and log that initializing
// Fetch remaining services and MEF components
// Switch to background thread
// Call `.SynchronizeWithServer`
// Call taintServerEventsListener.ListenAsync().Forget();
```

Could you try those changes too please?